src/commands/functions/list.js

```diff
@@ -56,40 +56,43 @@ class FunctionsListCommand extends Command {
       process.exit(1)
     }
 
-    const functions = getFunctions(functionsDir)
-    const functionData = Object.entries(functions)
+    const functions = await getFunctions(functionsDir)
 
-    if (functionData.length === 0) {
+    if (functions.length === 0) {
       this.log(`No functions found in ${functionsDir}`)
       this.exit()
     }
 
     if (flags.json) {
-      const jsonData = functionData.map(([functionName, { moduleDir }]) => {
-        const isDeployed = deployedFunctions.map((deployedFunction) => deployedFunction.n).includes(functionName)
-        return {
-          name: functionName,
-          url: `/.netlify/functions/${functionName}`,
-          moduleDir,
```

Comment:
This removes `moduleDir`. `moduleDir` is meant to be the site's `package.json` or the function's `package.json`.
Since `package.json` inside Netlify functions is not supported out-of-the-box, I don't think we should expose this to users. This could be a breaking change is anyone is using `--json` and `moduleDir`, i.e. would require a major release. What do you think?
We could put this in a separate PR.

Reply:
I think we're good here due to https://github.com/netlify/cli/blob/9292288cbb8b5b58aa4420a05c39795ba1fed474/src/commands/functions/list.js#L118